XLS-0081d-permissioned-dex/README.md

```diff
@@ -0,0 +1,592 @@
+</details>
+
+We propose these additions:
+
+| Field Name | Required? | JSON Type | Internal Type | Description |
+|------------|-----------|-----------|---------------|-------------|
+|`DomainID`| |`string`|`Hash256`|The domain that the offer must be a part of.|
+
+#### 5.1.1. `DomainID`
+
+The `DomainID` can only be included if the payment is a cross-currency payment (i.e. if the payment is going to interact with the DEX). It should only be included if the payment is permissioned.
+
+#### 5.1.2. `Flags`
+
+This spec proposes an additional flag, to indicate whether the offer should consider only the `DomainID`'s Permissioned DEX or also include the open DEX.
+
+### 5.2. Failure Conditions
+
+The existing set of failure conditions for `Payment` will continue to exist.
+
+There will also be the following in addition, if the `DomainID` is included:
+* The payment isn't a cross-currency payment.
+* The domain doesn't exist.
+* The `Account` is not a domain member.
+* The paths do not satisfy the domain's rules (e.g. a path includes an account that isn't a part of the domain).
+
+### 5.3. State Changes
+
+The existing set of state changes for a successful `Payment` transaction will continue to exist.
+
+## 6. RPC: `book_offers`
+
+The [`book_offers` RPC method](https://xrpl.org/book_offers.html) already exists on the XRPL. This proposal suggests some modifications to also support permissioned DEX domains.
+
+### 6.1. Request Fields
+
+As a reference, here are the fields that `book_offers` currently accepts:
+
+| Field Name | Required? | JSON Type | Description |
+|------------|-----------|-----------|-------------|
+|`taker_gets`|✔️|`object`|The asset the account taking the Offer would receive, as a currency without an amount.|
+|`taker_pays`|✔️|`object`|The asset the account taking the Offer would pay, as a currency without an amount.|
+|`ledger_hash`| |`string`|A 20-byte hex string for the ledger version to use.|
+|`ledger_index`| |`number` or `string`|The ledger index of the ledger to use, or a shortcut string to choose a ledger automatically.|
+|`limit`| |`number`|The maximum number of Offers to return. The response may include fewer results.|
+|`taker`| |`string`|The address of an account to use as a perspective. The response includes this account's offers even if they are unfunded. (You can use this to see what offers are above or below yours in the order book.)|
+
+This proposal puts forward the following addition:
+
+| Field Name | Required? | JSON Type | Description |
+|------------|-----------|-----------|-------------|
+|`domain`| |`string`|The object ID of a `PermissionedDomain` object. If this field is included, then the offers will be filtered to only show the valid domain offers for that domain.|
+
+### 6.2. Response Fields
+
+This proposal does not suggest any changes to the response fields. As a reference, here are the fields that `book_offers` currently returns:
```

Comment:
It should return `domain` somewhere - or in the root of the response or in the every offer objects itself.